### ISSUE TITLE
braille: show selection in edit controls and documents when tethered to review

### ISSUE THREAD

**Issue by burmancomp** [nvaccess/nvda#15994]
### Is your feature request related to a problem? Please describe.
Although "show selection" is enabled, NVDA does not show selection automatically in braille when braille is tethered to review. Fortunately pr #15844 and pr #15858 are available.
### Describe the solution you'd like
NVDA should show selection automatically when "show selection" is enabled.
### Describe alternatives you've considered
Some modifications to following functions:

- `ReviewTextInfoRegion._getSelection`
- `TextInfoRegion.nextLine`
- `TextInfoRegion.previousLine`
- `TextInfoRegion.update`
- `BrailleHandler.handleReviewMode`

`nextLine` and `previousLine` could be defined in `ReviewTextInfoRegion`. As to `update` I think defining it in `ReviewTextInfoRegion` is not so useful because new code needed is just some lines and calling it as super class function may require / likely requires it to be modified anyway.

**Comment by Jykke67** [comment]
Yes. The selected area is shown correctly in your test version of NVDA, so even the TC buttons. Therefore this feature should be implemented in future versions of NVDA.

**Comment by irrah68** [comment]
This would be a very useful feature in future releases of NVDA. I tried a test version compiled by @burmancomp and it seems to work. I always use Tether Braille to review setting and now it is possible to select text areas and see what is selected. The touch cursor keys work correctly. I can also navigate out of selected text and navigate back using Braille display's keys, and the selected text is shown unless I press a touch cursor or — for example — the arrow keys of the keyboard. I tested this in browse mode and in Notepad++ text editor.

**Comment by michaelDCurran** [comment]
Braille was originally designed so that only one cursor (expanded or collapsed) would be shown on the display. E.g. when tethered to review, you would only see the review cursor. When tethered to focus, yoyou would only see caret / selection (expanded caret).  I would bee curious to see any prototype of this idea so that I can get a better idea of what the UX would feel like.

**Comment by burmancomp** [comment]
Solution has limitations but on the other hand tethering braille to review gives more possibilities, and showing selection should be one of them. Can I open pull request for prototype?

**Comment by seanbudd** [comment]
Feel free to open a draft PR

**Comment by Jykke67** [comment]
Even I can’t see any drawbacks in this feature.

I want to know that is really selected and what is unselected during navigation like the sighted want to know.

In JAWS even selected desktop items are shown with dots 7 and 8. When unselected, the same items are shown without dots 7 and 8. Would be very confusing if they would always be shown as selected.

When typing this I know that everybody doesn’t prefer this, and it is an user option.

And yes, when navigating around a selected item with JAWS, the item remains selected and is shown with dots 7 and 8 and the unselected ones without dots 7 and 8.

Have tried to explain this as clearly as possible. Don’t know if succeeded.
This is just for comparison.

**Comment by burmancomp** [comment]
Do I close this issue?

**Comment by Adriani90** [comment]
No, this is still valid. The solution though needs a redesign as documented in the closed PR.

**Comment by burmancomp** [comment]
I have another version which does not depend on scripts and `_getSelection` just returns TextInfo. Logic what it returns is in helper function and `update` function.

If decision is that this feature requires anyway larger redesign then I do not open new pr.

**Comment by seanbudd** [comment]
@burmancomp - it is hard to provide feedback without more information on the approach. I would encourage either a more clear explanation or a draft PR again
